luarules/gadgets.lua

```diff
@@ -1123,6 +1129,9 @@ function gadgetHandler:Shutdown()
 end
 
 function gadgetHandler:GameFrame(frameNum)
+	-- Since GameGrame should never be called nested ensure here the callinDepth
+	-- is ok. We set it to 1 so after the run it will be set to 0 again.
+	callinDepth = 1
```

Comment:
Why? Isn't this redundant? It should never be the case that the value isn't already 1 at this point. If there is a worry that `callinDepth = callinDepth - 1` can get skipped due to error within a gadget then perhaps the call should be wrapped inside `pcall` or similar.

Reply:
I wanted to avoid doing a pcall inside every callin, this way it can maintain consistency by max next frame while avoiding checks, I think it's a good compromise since also reentrant callins doing gadget reordering are quite rare atm, I wanted to avoid overhead. Anyways if you think it's better I can do the pcall way.

Also want to do some profiling with the different checks to see if it really affects perf or not but for now I thought this is safest compromise.